src/lib/bit/bit.c

```diff
@@ -332,3 +332,110 @@ bit_count(const uint8_t *data, size_t bit_offset, size_t length)
 
 	return count;
 }
+
+void
+bit_copy_range_reverse(uint8_t *restrict dst, size_t dst_i,
+		       const uint8_t *restrict src, size_t src_i, size_t count)
+{
+	if (count == 0)
+		return;
+
+	assert(src_i >= count - 1);
+	/* Assure the buffers are non-overlapping. */
+	assert(src + (src_i - count + 1) / CHAR_BIT >
+	       dst + (dst_i + count) / CHAR_BIT ||
+	       dst + dst_i / CHAR_BIT > src + src_i / CHAR_BIT);
+	/*
+	 * The code structure is same as in the forward copy. Note that
+	 * bit position in source is counted from the byte end. This
+	 * allows to follow same logic as in the forward case.
+	 */
+	size_t dst_i_byte = dst_i / CHAR_BIT;
+	size_t dst_i_bit = dst_i % CHAR_BIT;
+	size_t src_i_byte = src_i / CHAR_BIT;
+	size_t src_i_bit = CHAR_BIT - 1 - src_i % CHAR_BIT;
```

Comment:
We should probably rename this to something like `src_tail_size` to reflect its actual meaning, preferably with introduction of a true `src_i_bit = src_i % CHAR_BIT`. This should make its explanation easier.